### PR TITLE
Clif fixes

### PR DESCRIPTION
**Pull request by hemagx**

### CODE REVIEW

File: src/map/clif.c [hunk 1]
```diff
@@ -267,19 +267,39 @@ uint32 clif_refresh_ip(void)
 
 #if PACKETVER >= 20071106
 static inline unsigned char clif_bl_type(struct block_list *bl) {
+	struct view_data *vd;
 	nullpo_retr(0x1, bl);
+
+	vd = status->get_viewdata(bl);
```
Comment (4144) [inline src/map/clif.c]: I think need add nullpo_retr(CLUT_NPC, vd);
Because if vd is NULL this is error.

File: src/map/clif.c [hunk 2]
```diff
@@ -267,19 +267,39 @@ uint32 clif_refresh_ip(void)
 
 #if PACKETVER >= 20071106
 static inline unsigned char clif_bl_type(struct block_list *bl) {
+	struct view_data *vd;
 	nullpo_retr(0x1, bl);
```
Comment (4144) [inline src/map/clif.c]: Here need nullpo_retr(CLUT_NPC, bl);

File: src/map/clif.c (hunk 1 above)
Comment (4144) [inline src/map/clif.c]: And call to status->get_viewdata need do from case

Comment (hemagx) [inline src/map/clif.c]: we can't do a nullpo check on vd as it's not always used and maybe can be null in cases it's not used, check rest of he code.

File: src/map/clif.c (hunk 2 above)
Comment (hemagx) [inline src/map/clif.c]: woah a leftover :D

**Pull request merged**